src/Components/Server/src/Circuits/CircuitOptionsJavaScriptInitializersConfiguration.cs

```diff
@@ -0,0 +1,33 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Text.Json;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.AspNetCore.Components.Server.Circuits
+{
+    internal class CircuitOptionsJavaScriptInitializersConfiguration : IConfigureOptions<CircuitOptions>
+    {
+        private readonly IWebHostEnvironment _environment;
+
+        public CircuitOptionsJavaScriptInitializersConfiguration(IWebHostEnvironment environment)
+        {
+            _environment = environment;
+        }
+
+        public void Configure(CircuitOptions options)
+        {
+            var file = _environment.WebRootFileProvider.GetFileInfo($"{_environment.ApplicationName}.modules.json");
+            if (file.Exists)
+            {
+                var initializers = JsonSerializer.Deserialize<string[]>(file.CreateReadStream());
+                for (var i = 0; i < initializers.Length; i++)
+                {
+                    var initializer = initializers[i];
+                    options.JavaScriptInitializers.Add(initializer);
```

Comment:
For 2 reaons:
* It doesn't get generated when there are no modules, and want to avoid the 404.
* The manifest includes the app name, which would need to be passed down to JS otherwise. We don't know if the manifest is for a blazor server app or if an MVC app wants to consume it (that's our contract for MVC/RP) so the endpoint abstracts over that.

I tried to do it through the websocket connection, but the ordering in which things happen makes it problematic.

Reply:
The 404 reason makes sense. Thanks.

Presumably people could actually fetch `AppName.manifest.json` (or whatever it's called) if they knew the URL to it within `wwwroot`, but we don't really care before it's only the same info they can get through the middleware anyway.